### DISCUSSION TITLE
Custom Reports

### DISCUSSION THREAD

**Discussion by Mbuso-Harvey** [frePPLe/frepple#398] [Q&A]
Hi Everyone, 

Just wondering if anybody has used the custom reports feature in Frepple? Looking for any resources in the form tutorials. Would appreciate if you  could point me in the right direction. 

Thanks

Harvey

**Comment by jdetaeye** [comment]
All you need to do is to write an SQL statement on the database to pull the data you need.  It'll require an SQL-savvy IT'er to know SQL and peek into the frepple database structure to write the query.

There isn't any detailed tutorial on how to do this...

**Comment by Mbuso-Harvey** [comment]
Thanks. On my interface the headings circled in the image aren't appearing.
So I wanted to test the query but cannot do so.

What do I do to have them appear?

Best

Harvey

**Reply by Mbuso-Harvey** [reply]
Wanted to attach the screenshot, but couldn't. The headings I am taking about are "Save", "Test", "Format" and "Delete".

**Reply by mike-ivc-inform** [reply]
the problem is not very clear, if you attach a screenshot, it may make it easier to understand. However, if I correctly understand your problem, then the role of report_role is not connected to you, so there may not be enough permissions to perform operations with the report. You need to add permissions (GRANT SELECT ON TABLE public....... TO report_role;)  to the tables that you use in the report .

<img width="803" alt="img0120" src="https://user-images.githubusercontent.com/87300645/132915158-824ba0e2-87f6-4849-9c2f-1119905ee69e.png">
<img width="559" alt="img0121" src="https://user-images.githubusercontent.com/87300645/132915162-5cd79f7a-4fed-4f4d-97cc-a55063ec0978.png">
<img width="604" alt="img0122" src="https://user-images.githubusercontent.com/87300645/132915165-ac0de893-6701-48c7-b9e0-3deb1d831db6.png">
<img width="938" alt="img0123" src="https://user-images.githubusercontent.com/87300645/132915169-c5e77233-a837-4782-9f46-a51c4516d677.png">
